src/dev/i18n/extract_default_translations.js

```diff
@@ -73,25 +114,40 @@ export async function extractDefaultTranslations(inputPath) {
       [hbsEntries, extractHandlebarsMessages],
     ].map(async ([entries, extractFunction]) => {
       const files = await Promise.all(
-        entries.map(async entry => {
-          return {
-            name: entry,
-            content: await readFileAsync(entry),
-          };
-        })
+        entries
+          .filter(entry => !config.exclude.includes(normalize(path.relative('.', entry))))
+          .map(async entry => {
+            return {
+              name: entry,
+              content: await readFileAsync(entry),
+            };
+          })
       );
 
       for (const { name, content } of files) {
         try {
           for (const [id, value] of extractFunction(content)) {
-            addMessageToMap(defaultMessagesMap, id, value);
+            validateMessageNamespace(id, name);
+            addMessageToMap(targetMap, id, value);
           }
         } catch (error) {
           throw new Error(`Error in ${name}\n${error.message || error}`);
         }
       }
     })
   );
+}
+
+export async function extractDefaultTranslations({ paths, output }) {
+  const defaultMessagesMap = new Map();
+
+  for (const inputPath of await getPluginsPaths(paths)) {
+    await extractMesssagesFromPathToMap(inputPath, defaultMessagesMap);
+  }
+
+  if (!output || !defaultMessagesMap.size) {
```

Comment:
__nit:__ please add a brief comment here so that our intention is clearer (like if `output` isn't supplied then we don't write into file or something along these lines).